c/polyjuice.h

```diff
@@ -701,6 +701,51 @@ void emit_log(struct evmc_host_context* context, const evmc_address* address,
   return;
 }
 
+/**
+ * check address collision
+ * check existence of eth_addr
+ * - if exists, then check nonce > 0 or it's NOT an EOA: return AddressCollision error
+ * - else, we can overwrite addr mapping
+ * @param overwrite true if there is a collision but we can continue to create a new account
+ * @return 0 means success
+*/
+int check_address_collision(gw_context_t* ctx, const uint8_t eth_addr[ETH_ADDRESS_LEN], bool* overwrite) {
+  gw_reg_addr_t addr = new_reg_addr(eth_addr);
+  uint8_t script_hash[32] = {0};
+  int ret = ctx->sys_get_script_hash_by_registry_address(ctx, &addr, script_hash);
+  if (ret == GW_ERROR_NOT_FOUND) {
+    return 0;
+  }
+  if (ret != 0) {
+    return ret;
+  }
+  uint32_t account_id;
+  ret = ctx->sys_get_account_id_by_script_hash(ctx, script_hash, &account_id);
+  if (ret != 0) {
+    return ret;
+  }
+  // account exists
+  uint32_t nonce;
+  ret = ctx->sys_get_account_nonce(ctx, account_id, &nonce);
+  if (ret != 0) {
+    return ret;
+  }
+  uint8_t code[MAX_DATA_SIZE];
```

Comment:
We could get the account_script by account_id.
And then, the account_script will tell us if it is an EOA.
By this way, we may avoid the cost of load_account_code.

Reply:
Are you talking about invoking `load_account_script` directly? I can see `load_account_code` invokes `load_account_script`. Looks like it just does what you say.

Reply:
yeap, you are right.